mm2src/coins/utxo.rs

```diff
@@ -500,6 +497,50 @@ impl UtxoCoinFields {
             hash_algo: self.tx_hash_algo.into(),
         }
     }
+
+    pub fn check_withdraw_address_supported(&self, addr: &Address) -> Result<(), MmError<UnsupportedAddr>> {
+        let conf = &self.conf;
+
+        match addr.addr_format {
+            // Considering that legacy is supported with any configured formats
+            // This can be changed depending on the coins implementation
+            UtxoAddressFormat::Standard => {
+                let is_p2pkh = addr.prefix == conf.pub_addr_prefix && addr.t_addr_prefix == conf.pub_t_addr_prefix;
+                let is_p2sh = addr.prefix == conf.p2sh_addr_prefix
+                    && addr.t_addr_prefix == conf.p2sh_t_addr_prefix
+                    && conf.segwit;
+                if !is_p2pkh && !is_p2sh {
+                    MmError::err(UnsupportedAddr::PrefixError)
+                } else {
+                    Ok(())
+                }
+            },
+            UtxoAddressFormat::Segwit => {
+                if !conf.segwit {
+                    return MmError::err(UnsupportedAddr::SegwitNotActivated(conf.ticker.clone()));
+                }
+
+                if addr.hrp != conf.bech32_hrp {
+                    MmError::err(UnsupportedAddr::HrpError {
+                        ticker: conf.ticker.clone(),
+                        hrp: addr.hrp.clone().unwrap_or_default(),
+                    })
+                } else {
+                    Ok(())
+                }
+            },
+            UtxoAddressFormat::CashAddress { .. } => {
+                if addr.addr_format == conf.default_address_format || addr.addr_format == self.my_address.addr_format {
+                    Ok(())
+                } else {
+                    MmError::err(UnsupportedAddr::LegacyActivated {
```

Comment:
It won't always be `LegacyActivated`, this else part will be also reached if an invalid cash address prefix is used. I think there are two options:
1. Rename the enum variant.
2. Add more enum variant(s) and return here when appropriate.

Reply:
other errors  will be returned from `address_from_str_unchecked` first https://github.com/KomodoPlatform/atomicDEX-API/blob/a2ee99c3ca23ca7703b5cda0dabaabacee607117/mm2src/coins/utxo/utxo_common.rs#L242-L244
for instance invalid cash address prefix will be returned from here  https://github.com/KomodoPlatform/atomicDEX-API/blob/a2ee99c3ca23ca7703b5cda0dabaabacee607117/mm2src/mm2_bitcoin/keys/src/address.rs#L212 but to make the error more general It can be changed to `InvalidFormatActivated` instead of `LegacyActivated`